### ISSUE TITLE
Optional Configurable energy cost for commands

### ISSUE THREAD

**Issue by Zhab80** [hilburn/AdvancedSystemsManager#2]
Hi, I'm a fan of the original mod which inspired this rewrite (SFM). I've spent many hours tinkering with programs with a big smile in the original. But the original have an issue which I believe is holding that mod back. The complete lack of any energy/fuel cost whatsoever.

As a modpack developer who likes to create challenging pack where everything comes with a price and where effective designs are rewarded, "SFM" as it is is simply too powerful and too forgiving of sloppy designs. As a server admin, this is heavy as players will often run poorly designed CPU hungry programs several time per second 24/7... and why not ? It's 100% free right ?

I gave a lot of consideration about how to implement energy costs to best solve the issues noted above. What seems best is having each command block cost a bit of energy every time they are executed. Small programs and/or programs that run infrequently would be less expensive than huge programs  and/or programs that run several times per seconds 24/7.

If each commands cost energy, this would strongly encourage people to come up with efficient designs instead of whatever the hell just works somehow however poorly design and wasteful it might be. It would also strongly motivate players to run programs as little as possible instead of 10 time per seconds forever. For example there is no need to check the growth state of crops 10 times per seconds. Checking once every 10 seconds would be 100 times less expensive and once every 30 seconds would be 300 times less expensive. You can use a pressure plate to activate an inventory relay once when a player is actually there instead of checking for a player 10 times per seconds 24/7. If the check cost 10 Rf and the programs run 10 times per seconds 24/7, that would be 8.64 million Rf per day... saved with a pressure plate. Food for thought.

This would serves as a powerful balancing mechanism that would keep this mod in check when running side by side with other mods. It would reward good engineering and punish sloppy programming. Since efficiency would now actually matter, the mod would become more server friendly as players would no longer spam persistent programs all over the place.

Furthermore, this would encourage players to explore the ways to use the blocks without a manager (at least with the original, I don't know of this rewrite will have this feature). For examples (in SFM), both the block gate and item valve can be used by themselves in surprising ways. This is an under used feature of that mod which deserves more attention in my opinion. If the manager would require power to do stuff, it would be more tempting for players to try to achieve the same results with standalone blocks with no managers.

Thoughts ?

**Comment by Keridos** [comment]
:+1: for this one

**Comment by Zhab80** [comment]
Thank you for the support.

**Comment by bladedpenguin** [comment]
TLDR but +1 energy usage
